### PR TITLE
add absolute alias for abs

### PR DESCRIPTION
**Pull request by jes-bz**
Adds an absolute alias for the abs function to match Numpy's use of both:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.absolute.html

Adds test to ensure the output from abs and absolute are the same.

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -15458,6 +15458,13 @@ def test_abs_signed_zero(self, device, dtype):
         for v in inp:
             self.assertGreater(math.copysign(1.0, v), 0.0)
 
+    @dtypes(torch.float)
+    def test_absolute(self, device, dtype):
+        # absolute is an alias for abs. Just check to see that results
+        # are the same.
+        t = torch.randn(10, 10, device=device, dtype=dtype)
+        self.assertEqual(t.abs(), t.absolute())
```
Comment (mruberry) [inline test/test_torch.py:15466]: I neglected to mention this earlier, but let's test the variants (out=, t.absolute(), and t.absolute_) too

**Comment by eellison** [comment]
This is a good change, however it would be nice if there was a mechanism to register an alias for an operator. This PR duplicates logic for `abs` and makes it harder to for downstream consumers of the IR, such as TorchScript passes. E.g. guard elimination (and therefore codegen) will now work for `torch.abs` but not for `torch.absolute`: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/guard_elimination.cpp#L261

cc @jamesr66a pytorch as a platform. One solution is a canonicalization pass, and another is just to only emit `aten::abs` in the IR from the start.

**Comment by mruberry** [comment]
An alias mechanism would be nice, especially since we'll likely have several more aliases in the future. Let's talk briefly offline about possible implementations?

**Comment by vadimkantorov** [comment]
NumPy has a bunch of back-compat legacy of itself e.g. numpy.diff vs older numpy.ediff1d and a related numpy.gradient. I hope PyTorch doesn't import all these older/quirky variants for numpy-compat...

**Comment by mruberry** [comment]
> NumPy has a bunch of back-compat legacy of itself e.g. numpy.diff vs older numpy.ediff1d and a related numpy.gradient. I hope PyTorch doesn't import all these older/quirky variants in the goal of numpy-compat...

np.absolute is the actual function name in NumPy, and np.abs is an alias for it. Don't worry, we're not implementing legacy NumPy names.

File: test/test_torch.py [hunk 2]
```diff
@@ -15458,6 +15458,27 @@ def test_abs_signed_zero(self, device, dtype):
         for v in inp:
             self.assertGreater(math.copysign(1.0, v), 0.0)
 
+    @dtypes(torch.float)
+    def test_absolute(self, device, dtype):
```
Comment (mruberry) [inline test/test_torch.py:15462]: Did you lose the torch.absolute(t) (with no out kwarg) variant?

Comment (jessebrizzi) [reply test/test_torch.py:15462]: It was never there, I can add it in.

Comment (mruberry) [reply test/test_torch.py:15462]: My mistake. Yep, let's add that just to be super thorough.

**Comment by vadimkantorov** [comment]
Then there also are `subtract`, `power`, `multiply` etc

File: torch/_tensor_docs.py [hunk 3]
```diff
@@ -173,6 +173,21 @@ def add_docstr_all(method, docstr):
...
+add_docstr_all('absolute',
+               r"""
+absolute() -> Tensor
+
+Alias for :func:`abs`
+""")
+
+add_docstr_all('absolute_',
+               r"""
+abssolute_() -> Tensor
```
Comment (mruberry) [inline torch/_tensor_docs.py:185]: typo: abs solute -> abs olute

File: test/test_torch.py [hunk 4]
```diff
@@ -15458,6 +15458,31 @@ def test_abs_signed_zero(self, device, dtype):
...
+        # absolute is an alias for abs. Just check to see that results
+        # are the same.
+        t1 = torch.randn(10, 10, device=device, dtype=dtype)
```
Comment (mruberry) [inline test/test_torch.py:15465]: Nittiest of nits: t and t_copy instead of t1 and t2

**Review by mruberry — APPROVED** [review]
Cool first addition to PyTorch, Jesse!

**Pull request closed without merging**